editor/editor_file_dialog.cpp

```diff
@@ -1730,6 +1786,11 @@ EditorFileDialog::EditorFileDialog() {
 	remove_dialog = memnew(DependencyRemoveDialog);
 	add_child(remove_dialog);
 
+	project_man_remove_dialog = memnew(ConfirmationDialog);
+	project_man_remove_dialog->set_text(TTR("Remove the selected files? For safety only files and empty directories can be deleted from here. (Cannot be undone.)\nDepending on your filesystem configuration, the files will either be moved to the system trash or deleted permanently."));
+	project_man_remove_dialog->connect("confirmed", callable_mp(this, &EditorFileDialog::_delete_files_project_man));
+	add_child(project_man_remove_dialog);
```

Comment:
I'm not sure I like the idea to have Project Manager specific code in EditorFileDialog. It's meant to be a general purpose file dialog, it would be better if there were relevant hooks for the Project Manager to configure the behavior that it wants (e.g. signals it can connect its own callbacks to).

Reply:
You're right, this is not how it should be. Fixed accordingly (see comment).